golang/cosmos/app/app.go

```diff
@@ -809,57 +824,87 @@ func normalizeModuleAccount(ctx sdk.Context, ak authkeeper.AccountKeeper, name s
 type cosmosInitAction struct {
 	Type        string             `json:"type"`
 	ChainID     string             `json:"chainID"`
+	IsBootstrap bool               `json:"isBootstrap"`
 	Params      swingset.Params    `json:"params"`
-	StoragePort int                `json:"storagePort"`
 	SupplyCoins sdk.Coins          `json:"supplyCoins"`
-	VibcPort    int                `json:"vibcPort"`
-	VbankPort   int                `json:"vbankPort"`
-	LienPort    int                `json:"lienPort"`
 	UpgradePlan *upgradetypes.Plan `json:"upgradePlan,omitempty"`
+	LienPort    int                `json:"lienPort"`
+	StoragePort int                `json:"storagePort"`
+	VbankPort   int                `json:"vbankPort"`
+	VibcPort    int                `json:"vibcPort"`
 }
 
 // Name returns the name of the App
 func (app *GaiaApp) Name() string { return app.BaseApp.Name() }
 
-func (app *GaiaApp) MustInitController(ctx sdk.Context) {
-	if app.controllerInited {
-		return
+func (app *GaiaApp) CheckControllerInited(expected bool) {
+	if app.controllerInited != expected {
+		fmt.Fprintf(os.Stderr, "controllerInited != %t\n", expected)
+		debug.PrintStack()
```

Comment:
Could you make sure that the stack is also printed to stderr?

Reply:
PrintStack does write to stderr:
```
// PrintStack prints to standard error the stack trace returned by runtime.Stack.
func PrintStack() {
	os.Stderr.Write(Stack())
}
```